### ISSUE TITLE
'Cannot open Failover Group' Error when connecting to Azure SQL through an Auto-failover group if the Initial Catalog prop on the connection string is empty

### ISSUE THREAD

**Issue by acomarce** [dotnet/SqlClient#1875]
### Describe the bug
When trying to connect to Azure SQL through an Auto-failover group listener hostname the ` Failed to connect to server tcp:db-01.database.windows.net,1433.` error occurs if the `InitialCatalog` property is set to an empty string.

When connecting directly to the database server that is part of the auto-failover group the error does not occur. 

**Note**: I'm not sure if this is an `SqlClient` issue or a `Microsoft.SqlServer.Management.Smo` issue, I will move it if necessary. 

### Exception
```
Exception message: Failed to connect to server tcp:db-01.database.windows.net,1433.

Stack trace:  icrosoft.SqlServer.Management.Common.ConnectionFailureException: Failed to connect to server tcp:db-01.database.windows.net,1433.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot open Failover Group "db-01.database.windows.net" requested by the login. The login failed.
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.TdsParser.Run(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj)
   at Microsoft.Data.SqlClient.SqlInternalConnectionTds.CompleteLogin(Boolean enlistOK)
   at Microsoft.Data.SqlClient.SqlInternalConnectionTds.AttemptOneLogin(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean ignoreSniOpenTimeout, TimeoutTimer timeout, Boolean withFailover)
   at Microsoft.Data.SqlClient.SqlInternalConnectionTds.LoginNoFailover(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString connectionOptions, SqlCredential credential, TimeoutTimer timeout)
   at Microsoft.Data.SqlClient.SqlInternalConnectionTds.OpenLoginEnlist(TimeoutTimer timeout, SqlConnectionString connectionOptions, SqlCredential credential, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance)
   at Microsoft.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling, String accessToken, DbConnectionPool pool)
   at Microsoft.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionFactory.CreateNonPooledConnection(DbConnection owningConnection, DbConnectionPoolGroup poolGroup, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at Microsoft.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at Microsoft.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry, SqlConnectionOverrides overrides)
   at Microsoft.Data.SqlClient.SqlConnection.Open(SqlConnectionOverrides overrides)
   at Microsoft.Data.SqlClient.SqlConnection.Open()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InternalConnectImpl()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InternalConnect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
ClientConnectionId:0c55c83c-2532-42e9-986f-2b9b475xxxxxxx
Error Number:40971,State:1,Class:20
   --- End of inner exception stack trace ---
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.PoolConnect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.GetServerInformation()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.get_DatabaseEngineEdition()
   at Microsoft.SqlServer.Management.Smo.ExecutionManager.GetDatabaseEngineEdition()
   at Microsoft.SqlServer.Management.Smo.Database.get_DatabaseEngineEdition()
   at Microsoft.SqlServer.Management.Smo.Database.GetPropertyMetadataProvider()
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.get_Properties()
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.AddObjectPropsFromDataReader(IDataReader reader, Boolean skipIfDirty, Int32 startColIdx, Int32 endColIdx)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.AddObjectPropsFromDataReader(IDataReader reader, Boolean skipIfDirty)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.Microsoft.SqlServer.Management.Sdk.Sfc.ISqlSmoObjectInitialize.InitializeFromDataReader(IDataReader reader)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.CreateNewObjectFromRow(AbstractCollectionBase childColl, Type childType, IDataReader reader, Int32 columnIdx, Boolean isOrderedByID, Boolean skipOrderChecking)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitObjectsFromEnumResultsRec(SqlSmoObject currentSmoObject, XPathExpression levelFilter, Int32 filterIdx, IDataReader reader, Int32 columnIdx, Object[] parentRow, Boolean forScripting, List`1 urnList, Int32 startLeafIdx)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitObjectsFromEnumResults(Urn levelFilter, IDataReader reader, Boolean forScripting, List`1 urnList, Int32 startLeafIdx, Boolean skipServer)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitChildLevel(Urn levelFilter, ScriptingPreferences sp, Boolean forScripting, IEnumerable`1 extraFields)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh, ScriptingPreferences sp, String filterQuery, IEnumerable`1 extraFields)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh, ScriptingPreferences sp)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.Refresh(Boolean refreshChildObjects)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.Refresh()
   at ConsoleApp1.Program.Main(String[] args) in C:\Users\Aleksandar.Markovski\RiderProjects\ConsoleApp1\ConsoleApp1\Program.cs:line 17
```

### To reproduce

See the further technical details bellow for the versions of the libraries used. 

Crude repro code :) : 
```c#
using Microsoft.Data.SqlClient;
using Microsoft.SqlServer.Management.Common;
using Microsoft.SqlServer.Management.Smo;

namespace ConsoleApp1;

internal static class Program
{
    static void Main(string[] args)
    {
        var dbServer = "";
        var dbName = "test";
        var dbUsername = "";
        var dbPassword = "";

        var cnnString =
            $"Server=tcp:{dbServer},1433;Initial Catalog={dbName};Persist Security Info=False;User ID={dbUsername}; Password={dbPassword};";

        var server = Build(cnnString);

        // Fails here
        server.Databases.Refresh();
    }

    public static Server Build(string connectionString)
    {
        var builder = new SqlConnectionStringBuilder(connectionString)
        {
            // Works if the following is not defined
            // But we need this to list the dbs already present on the server and are not connecting to a specific DB
            InitialCatalog = string.Empty
        };

        var sqlConnection = new SqlConnection(builder.ToString());
        var connection = new ServerConnection(sqlConnection);
        return new Server(connection);
    }
}
```

### Expected behavior
The `Refresh` call succeeds and we are able to list the existing DBs when connecting through the listener, the same as if it were connecting directly.

### Further technical details
.Net version: net6.0

Libraries: 
* Microsoft.Data.SqlClient: 3.0.0
* Microsoft.SqlServer.SqlManagementObjects: 161.46521.71

**Comment by JRahnama** [comment]
@acomarce we will get back to this issue after holidays.

**Comment by acomarce** [comment]
@JRahnama any update on this?

**Comment by acomarce** [comment]
@E-G-C for visibility.

**Comment by lcheunglci** [comment]
Have you tried on the latest version of MDS i.e. 5.0.1 and if this was still the case?

**Comment by acomarce** [comment]
@lcheunglci bumping the SqlClient to 5.0.1 results in the following exception: 

```
System.MissingMethodException: Method not found: 'Void Microsoft.Data.SqlClient.SqlConnectionStringBuilder.set_Encrypt(Boolean)'.
   at Microsoft.SqlServer.Management.Common.ConnectionSettings.GetConnectionString()
   at Microsoft.SqlServer.Management.Common.ConnectionSettings.get_ConnectionString()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InitSqlConnectionObject(Boolean setConnectionString, Boolean removeIntegratedSecurity)
   at Microsoft.SqlServer.Management.Common.ConnectionManager..ctor(IRenewableToken token, Boolean removeIntegratedSecurity)
   at Microsoft.SqlServer.Management.Common.ConnectionManager..ctor()
   at Microsoft.SqlServer.Management.Common.ServerConnection..ctor(String serverInstance)
   at Microsoft.SqlServer.Management.Common.ServerConnection.ConnectionFactory.CreateServerConnection(String connString, String initialCatalog, Boolean poolConn, IRenewableToken accessToken)
   at Microsoft.SqlServer.Management.Common.ServerConnection.ConnectionFactory.GetDatabaseConnection(String dbName, Boolean poolConnection, IRenewableToken accessToken)
   at Microsoft.SqlServer.Management.Common.ServerConnection.GetDatabaseConnection(String dbName, Boolean poolConnection)
   at Microsoft.SqlServer.Management.Smo.Database.get_DatabaseExecutionManager()
   at Microsoft.SqlServer.Management.Smo.Database.get_ExecutionManager()
   at Microsoft.SqlServer.Management.Smo.Database.get_DatabaseEngineEdition()
   at Microsoft.SqlServer.Management.Smo.Database.GetPropertyMetadataProvider()
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.get_Properties()
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.AddObjectPropsFromDataReader(IDataReader reader, Boolean skipIfDirty, Int32 startColIdx, Int32 endColIdx)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.AddObjectPropsFromDataReader(IDataReader reader, Boolean skipIfDirty)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.Microsoft.SqlServer.Management.Sdk.Sfc.ISqlSmoObjectInitialize.InitializeFromDataReader(IDataReader reader)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.CreateNewObjectFromRow(AbstractCollectionBase childColl, Type childType, IDataReader reader, Int32 columnIdx, Boolean isOrderedByID, Boolean skipOrderChecking)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitObjectsFromEnumResultsRec(SqlSmoObject currentSmoObject, XPathExpression levelFilter, Int32 filterIdx, IDataReader reader, Int32 columnIdx, Object[] parentRow, Boolean forScripting, List`1 urnList, Int32 startLeafIdx)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitObjectsFromEnumResults(Urn levelFilter, IDataReader reader, Boolean forScripting, List`1 urnList, Int32 startLeafIdx, Boolean skipServer)
   at Microsoft.SqlServer.Management.Smo.SqlSmoObject.InitChildLevel(Urn levelFilter, ScriptingPreferences sp, Boolean forScripting, IEnumerable`1 extraFields)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh, ScriptingPreferences sp, String filterQuery, IEnumerable`1 extraFields)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh, ScriptingPreferences sp)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.InitializeChildCollection(Boolean refresh)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.Refresh(Boolean refreshChildObjects)
   at Microsoft.SqlServer.Management.Smo.SmoCollectionBase.Refresh()
   at ConsoleApp1.Program.Main(String[] args) in C:\Users\a.m\RiderProjects\ConsoleApp2\ConsoleApp2\Program.cs:line 22
```
So I guess SMO needs to update the supported versions of SQLClient


But also bumping `Microsoft.SqlServer.SqlManagementObjects` to `170.11.0` resolved it, and I can query the databases. 

Did some testing, and it seems the issue was fixed in SMO, [161.48028.0](https://www.nuget.org/packages/Microsoft.SqlServer.SqlManagementObjects/161.48028.0).
